### PR TITLE
Catalyst: Add Tag to Catalyst2 TPL + Cleaning

### PR DESCRIPTION
**Pull request by Naapple**
- This change makes the Catalyst2 TPL point to a particular tag (v2.0.0) rather than master in the Kitware Catalyst repo. An extra argument needed to be added (CATALYST_WRAP_PYTHON) in the build of the TPL. All SEACAS ctests pass.
- Cleaned up our Catalyst API2 work following the suggestions made in our previous pull request: Catalyst API 2 IOSS Full Implementation #431

### CODE REVIEW

**Comment by Naapple** [comment]
I have read the CLA Document and I hereby sign the CLA

File: packages/seacas/libraries/ioss/src/catalyst/Iocatalyst_CatalystManager.C [hunk 1]
```diff
@@ -262,7 +262,7 @@ namespace Iocatalyst {
       errmsg << "Catalyst pipeline is not a multi-input pipeline";
       IOSS_ERROR(errmsg);
     }
-    auto name = p.catalystMultiInputPipelineName;
+    const auto name = p.catalystMultiInputPipelineName;
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/catalyst/Iocatalyst_CatalystManager.C:265]: Could this be `const auto &name`

File: packages/seacas/libraries/ioss/src/catalyst/Iocatalyst_CatalystManager.C [hunk 2]
```diff
@@ -280,7 +280,7 @@ namespace Iocatalyst {
...
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/catalyst/Iocatalyst_CatalystManager.C:283]: Same as above.

File: packages/seacas/libraries/ioss/src/catalyst/Iocatalyst_CatalystManager.C [hunk 3]
```diff
@@ -262,7 +262,7 @@ namespace Iocatalyst {
...
     for (auto cp : catPipes) {
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/catalyst/Iocatalyst_CatalystManager.C:266]: Should this be `auto &cp` similar to line 284 or maybe even `const auto &cp`?

**Review by gsjaardema — APPROVED** [review]

**Comment by ajpelle** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ajpelle** [comment]
recheck

**Pull request merged**